Source/WebCore/rendering/RenderLayerModelObject.cpp

```diff
@@ -286,6 +287,11 @@ TransformationMatrix* RenderLayerModelObject::layerTransform() const
 
 void RenderLayerModelObject::updateLayerTransform()
 {
+    if (style().offsetPath()) {
+        MotionPathHelper::setReferenceBoxForPathOperations(*this);
+        if (auto* cb = containingBlock())
+            view().frameView().layoutContext().markNeedingTransformUpdate(cb, this);
```

Comment:
You only need to call this for ray operations, right, so it's a waste to do this work for every renderer with offset path.

Reply:
Also you're going to go ahead and call updateTransform() here anyway, and then a second time later for things with ray operations?

Reply:
Right, added a function on `MotionPath` to specify if the `PathOperation` has a dependency on the containing block's size. We return early for this case so we don't calculate the transform twice.